tests/service/redshift: Implement Sweeper for Redshift Cluster Snapshots

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Description

The acceptance testing can leave lingering resources on failed runs (e.g. via coding issues during development or panics). These lingering resources can cause quota limits to be reached, extraneous money to be spent, or in some worst cases leave open security risks in the testing account.

We do not require sweeper implementation during initial resource implementation at this time, but this may change in the future when the process can be simplified and documented better.

These snapshots are being created by the `aws_redshift_cluster` resource `final_snapshot` handling, which now has a recurring test failure:

```
=== CONT  TestAccAWSRedshiftCluster_withFinalSnapshot
TestAccAWSRedshiftCluster_withFinalSnapshot: testing.go:745: Error destroying resource! WARNING: Dangling resources
may exist. The full state and error is shown below.
Error: errors during apply: Error deleting Redshift Cluster (tf-redshift-cluster-1769307370542232758): ClusterSnapshotQuotaExceeded: cannot create more than 20 user snapshots
```

However our typical convention for sweepers is to place the sweeping logic with its matching resource (so in this case, it would be `aws_redshift_cluster_snapshot`). While we don't have existing source code for a Redshift Cluster Snapshot resource to add the sweeper currently, a new test file can be created with just the new sweeper. This new sweeper should have a dependency on the `aws_redshift_cluster` resource sweeper.

### Definition of Done

* New `aws/resource_aws_redshift_cluster_snapshot_test.go` file created with new `aws_redshift_cluster_snapshot` sweeper that has dependency on existing `aws_redshift_cluster` resource sweeper
* Testing account empty of this resource:

```console
$ aws redshift describe-cluster-snapshots | jq '.Snapshots | length'
0
```

### References

* Prior art with `aws_redshift_cluster`: https://github.com/terraform-providers/terraform-provider-aws/blob/7448296d6141bda2878e17638de99b303aabb789/aws/resource_aws_redshift_cluster_test.go#L18-L60
* [Extending Terraform sweeper documentation](https://www.terraform.io/docs/extend/testing/acceptance-tests/sweepers.html)

--- a/aws/resource_aws_redshift_cluster_snapshot_test.go
+++ b/aws/resource_aws_redshift_cluster_snapshot_test.go
@@ -1,0 +1,62 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/redshift"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func init() {
+	resource.AddTestSweepers("aws_redshift_cluster_snapshot", &resource.Sweeper{
+		Name: "aws_redshift_cluster_snapshot",
+		F:    testSweepRedshiftClusterSnapshots,
+		Dependencies: []string{
+			"aws_redshift_cluster",
+		},
+	})
+}
+
+func testSweepRedshiftClusterSnapshots(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %w", err)
+	}
+	conn := client.(*AWSClient).redshiftconn
+
+	err = conn.DescribeClusterSnapshotsPages(&redshift.DescribeClusterSnapshotsInput{}, func(resp *redshift.DescribeClusterSnapshotsOutput, isLast bool) bool {
+		if len(resp.Snapshots) == 0 {
+			log.Print("[DEBUG] No Redshift cluster snapshots to sweep")
+			return false
+		}
+
+		for _, s := range resp.Snapshots {
+			id := aws.StringValue(s.SnapshotIdentifier)
+
+			if !strings.EqualFold(aws.StringValue(s.SnapshotType), "manual") || !strings.EqualFold(aws.StringValue(s.Status), "available") {
+				log.Printf("[INFO] Skipping Redshift cluster snapshot: %s", id)
+				continue
+			}
+
+			log.Printf("[INFO] Deleting Redshift cluster snapshot: %s", id)
+			_, err := conn.DeleteClusterSnapshot(&redshift.DeleteClusterSnapshotInput{
+				SnapshotIdentifier: s.SnapshotIdentifier,
+			})
+			if err != nil {
+				log.Printf("[ERROR] Failed deleting Redshift cluster snapshot (%s): %s", id, err)
+			}
+		}
+		return !isLast
+	})
+	if err != nil {
+		if testSweepSkipSweepError(err) {
+			log.Printf("[WARN] Skipping Redshift Cluster Snapshot sweep for %s: %s", region, err)
+			return nil
+		}
+		return fmt.Errorf("Error retrieving Redshift cluster snapshots: %w", err)
+	}
+	return nil
+}

